packages/resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -163,10 +163,8 @@ export class AwsEcsDetector implements ResourceDetector {
       );
       const splitData = rawData.trim().split('\n');
       for (const str of splitData) {
-        if (str.length > AwsEcsDetector.CONTAINER_ID_LENGTH) {
-          containerId = str.substring(
-            str.length - AwsEcsDetector.CONTAINER_ID_LENGTH
-          );
+        containerId = this._extractContainerIdFromLine(str);
+        if (containerId) {
```

Comment:
Looking at this again, this will try to match `Primary pattern -> Fallback -> Legacy Fallback` patterns, for each `str of splitData`. However, this will return after the first match (e.g. splitData[0] works with Legacy Fallback and would be return, instead splitData[1] that could have matched with Primary pattern).

I suppose what you really want is to change the logic to be something like:
```ts
for (const str of splitData) {
  // check Primary pattern, return if found
}
for (const str of splitData) {
  // check Fallback pattern, return if found
}
for (const str of splitData) {
  // check Legacy Fallback pattern, return if found
}
```